p2p/discover/table.go

```diff
@@ -502,6 +521,22 @@ func (tab *Table) addSeenNode(n *node) {
 	}
 }
 
+func (tab *Table) filterNode(n *node) bool {
+	if tab.enrFilter != nil {
```

Comment:
I would invert the if condition following an early exit approach. i.e.: `if tab.enrFilter == nil { return false }` this way the logic block will not be indented anymore which increases code readability